File: JavaScript/1_tower-of-hanoi.js
'use strict'

const readline = require('readline')
const rl = readline.createInterface({ input: process.stdin })

rl.on('line', input => {
    const a = 'A', b = 'B', c = 'C'
    hanoi(input, a, b, c)
    rl.close()
})

function hanoi(n, a, b, c) {
    if (!n) {
        return
    }
    hanoi(n - 1, a, c, b)
    console.log(a, '->', c)
    hanoi(n - 1, b, a, c)
}

File: Kotlin/1_tower-of-hanoi.kt
fun main() {
    val n: Int = readLine()!!.toInt()
    val a = 'A'
    val b = 'B'
    val c = 'C'
    hanoi(n, a, b, c)
}

fun hanoi(n: Int, a: Char, b: Char, c: Char) {
    if (n == 0) {
        return
    } else {
        hanoi(n - 1, a, c, b)
        println("$a -> $c")
        hanoi(n - 1, b, a, c)
    }
}
